packages/deploy-tooling/src/base/deploy.ts

```diff
@@ -281,6 +326,48 @@ async function getAbapService(config: AbapDeployConfig, logger: Logger): Promise
     return service;
 }
 
+/**
+ * Creates a new transport request using adt service.
+ *
+ * @param config - deployment configuration
+ * @param logger - reference to the logger instance
+ * @param provider - instance of the axios-extension abap service provider (it is created if not passed in)
+ * @returns transportRequest
+ */
+export async function createTransportRequest(
+    config: AbapDeployConfig,
+    logger: Logger,
+    provider?: AbapServiceProvider
+): Promise<string | void> {
+    if (!provider) {
+        provider = await getAbapServiceProvider(config, logger);
+    }
+
+    const adtService = await provider.getAdtService<TransportRequestService>(TransportRequestService);
+    if (adtService) {
+        const createTransportParams = {
+            packageName: config.app.package ?? '',
+            ui5AppName: config.app.name,
+            description: 'Created by @sap-ux/deploy-tooling'
+        };
+
+        try {
+            const transportRequest = await adtService.createTransportRequest(createTransportParams);
+            if (transportRequest) {
+                config.app.transport = transportRequest;
```

Comment:
Can we simply return the TR, it is not expected that the config is updated based on the jsdoc?

Reply:
I've updated to just return the TR and assign it to the config in the consuming function